### DISCUSSION TITLE
Automation of SQL Server patching through DBA Tool

### DISCUSSION THREAD

**Discussion by sharmakneeraj4** [dataplat/dbatools#7345] [General]
Hi Everyone,

I am trying to automate SQL Server patching through DBA tool but when I am looking at SQL Server 2012 all new updates are coming as security updates. Is there a way to install security updates that are not a part of CU or service pack through DBATool. I believe, there are some changes required in the parameter version. I want to update latest KB available that is KB4583465 SQL Version -11.0.7507.2

$version = '2014SP4'
 
 Update-DbaInstance -ComputerName $Servers -Path '\\xyz\SQL2012\CU\' -Credential $cred -Version $version -Restart -Confirm:$false

Also, I want this script to generate a log that can tell if there is any failure in the script. Can someone please suggest how to do that.

**Comment by niphlod** [comment]
@nvarscar can surely chime in, but you may want to use the -KB param to install whatever KB you want

**Reply by nvarscar** [reply]
Yup, that's exactly it. You need to use the -KB parameter.

**Reply by sharmakneeraj4** [reply]
Thanks for reply. so it should be like this?

$version = '2014SP4-KB'

**Reply by sharmakneeraj4** [reply]
Oh  I got it. I should be specified as a parameter 
$kb=''

Update-DbaInstance -ComputerName $Servers -Path '\xyz\SQL2012\CU' -Credential $cred -Kb $kb -Restart -Confirm:$false

Thanks a lot to both of you.

**Reply by niphlod** [reply]
if it's all that you needed remember to mark the question as answered
